worker/caasoperator/initializer.go

```diff
@@ -107,29 +107,26 @@ func (p initializeUnitParams) Validate() error {
 }
 
 // reTrier is used for re-running some certain retryable exec request.
-type reTrier func(func() error, Logger, clock.Clock, <-chan struct{}) error
-
-// runnerWithRetry retries exec requests for init unit process if it got 137 error.
-func runnerWithRetry(f func() error, logger Logger, clk clock.Clock, cancel <-chan struct{}) error {
+type reTrier func(func() error, func(error) bool, Logger, clock.Clock, <-chan struct{}) error
 
+// runnerWithRetry retries the exec request for init unit process if it got a retryable error.
+func runnerWithRetry(f func() error, fatalChecker func(error) bool, logger Logger, clk clock.Clock, cancel <-chan struct{}) error {
 	do := func() error {
-		if wrench.IsActive("exec", "137") {
-			fakeErr := errors.New("fake 137")
-			logger.Warningf("wrench exec 137 enabled, returns %v", fakeErr)
+		if wrench.IsActive("exec", "retryable-error") {
+			fakeErr := errors.New("fake retryable-error")
+			logger.Warningf("wrench exec retryable-error enabled, returns %v", fakeErr)
 			return exec.NewExecRetryableError(fakeErr)
 		}
 		return f()
 	}
 	args := retry.CallArgs{
-		Attempts:    5,
-		Delay:       1 * time.Second,
-		MaxDuration: 30 * time.Second,
-		Clock:       clk,
-		Stop:        cancel,
-		Func:        do,
-		IsFatalError: func(err error) bool {
-			return err != nil && !exec.IsExecRetryableError(err)
-		},
+		Attempts:     5,
+		Delay:        1 * time.Second,
+		MaxDuration:  30 * time.Second,
```

Comment:
I think we may need to consider using different values depending on what's being run. ie pass in as params
We can do some tests on AKS, GKE, microk8s under load etc to see

Reply:
the point to have this func is to share the retry config for retries inside remote process.
I don't think we need a param here or we just remove this func and call `retry.Call(retry.CallArgs{})` anywhere directly.